% !TEX TS-program = pdflatex
% !TEX encoding = UTF-8 Unicode

% This is a simple template for a LaTeX document using the "article" class.
% See "book", "report", "letter" for other types of document.

\documentclass[11pt]{article} % use larger type; default would be 10pt

%\usepackage[utf8]{inputenc} % set input encoding (not needed with XeLaTeX)


%%% Examples of Article customizations
% These packages are optional, depending whether you want the features they provide.
% See the LaTeX Companion or other references for full information.

%%% PAGE DIMENSIONS
\usepackage{geometry} % to change the page dimensions
\geometry{a4paper} % or letterpaper (US) or a5paper or....
\geometry{margin=1in} % for example, change the margins to 2 inches all round
% \geometry{landscape} % set up the page for landscape
%   read geometry.pdf for detailed page layout information

\usepackage{graphicx} 
% \usepackage[parfill]{parskip} % Activate to begin paragraphs with an empty line rather than an indent

%%% PACKAGES
%\usepackage{booktabs} % for much better looking tables
\usepackage{array} % for better arrays (eg matrices) in maths
%\usepackage{paralist} % very flexible & customisable lists (eg. enumerate/itemize, etc.)
\usepackage{verbatim} % adds environment for commenting out blocks of text & for better verbatim
%\usepackage{subfig} % make it possible to include more than one captioned figure/table in a single float
% These packages are all incorporated in the memoir class to one degree or another...

%%% HEADERS & FOOTERS
\usepackage{fancyhdr} % This should be set AFTER setting up the page geometry
\pagestyle{fancy} % options: empty , plain , fancy
\renewcommand{\headrulewidth}{0pt} % customise the layout...
\lhead{}\chead{}\rhead{}
\lfoot{}\cfoot{\thepage}\rfoot{}

%%% SECTION TITLE APPEARANCE
%\usepackage{sectsty}
%\allsectionsfont{\sffamily\mdseries\upshape} % (See the fntguide.pdf for font help)
% (This matches ConTeXt defaults)

%%% ToC (table of contents) APPEARANCE
%\usepackage[nottoc,notlof,notlot]{tocbibind} % Put the bibliography in the ToC
%\usepackage[titles,subfigure]{tocloft} % Alter the style of the Table of Contents
%\renewcommand{\cftsecfont}{\rmfamily\mdseries\upshape}
%\renewcommand{\cftsecpagefont}{\rmfamily\mdseries\upshape} % No bold!

%\usepackage[T1]{fontenc}
\usepackage[latin9]{inputenc}
%\usepackage[active]{srcltx}
\usepackage{setspace}
\doublespacing
\usepackage[english]{babel}

\begin{document}

\title{Connectivity Analysis of Metagenomic Data}
\author{ACH, JP, RCK, RM, JJ, JMT, CTB}
\maketitle

\section{Introduction}
Given the rapid decrease in the costs of sequencing, we can now achieve the sequencing depth necessary to study even the most complex environments \cite{Hess:2011p686,Qin:2010p189}.  The main bottleneck for these metagenomic studies is the lack of effective strategies to annotate and predict gene functions from the enormous sequencing datasets that are now being generated \cite{Hoff:2009p913,Kunin:2008p16,Noguchi:2006p968,Zhang:2012p959}.  De novo metagenomic assembly has been used as a solution to reduce dataset size by collapsing numerous short reads into fewer contigs and providing longer sequences containing multiple genes and operons \cite{Miller:2010p226,Pop:2009p798}.  Furthermore, because it does not rely on the availability of reference genomes, assembly produces novel contigs which can be compared within and between metagenomes \cite{Li:2009p707,Schloss:2008p2}.  With sufficient sequencing, novel draft genomes can also be obtained from these assemblies \cite{Hess:2011p686}.  The application of de novo assembly to large and complex metagenomic datasets relies on the ability to store information about the connectivity of sequencing reads within an assembly graph.  Thus, metagenomic de novo assembly is limited by both the diversity of the metagenome and the availability of computational memory.  

Recently developed metagenome-specific de novo assemblers use various "divide and conquer" approaches to break apart components of the assembly graph \cite{Peng:2011p898} (cite metaVelvet), taking advantage of the fact that environmental populations contain multiple genomes which have been sampled at varying depths corresponding to their natural abundance.  Read coverage and/or graph connectivity are used to break apart and simplify metagenomic assembly graph into subcomponents which are subsequently assembled separately.  In order to resolve these components of an assembly graph, variable-coverage sequences must accurately be distinguished from from sequencing errors and bias.  The presence of sequencing biases and errors have been demonstrated in Illumina sequences \cite{Harismendy:2009p228,Hoffmann:2009p1027,Nakamura:2011p741} but very little is known about their effects on assembly graph properties and the resulting assemblies.  With large amounts of sequencing (as is needed for complex metagenomes), increases in the number of real biological sequences are accompanied by increases in sequencing errors and biases.  In this study, we analyzed the effects of errors and sequencing biases on assembly graphs, and our ability to resolve disconnected components of several complex metagenomic assembly graphs.  We identified highly-connecting sequences in several metagenomes which we demonstrate originate, at least partially, from sequencing artifacts.  We evaluated the effects of removing these sequences on metagenomic assembly and discuss how this approach ultimately enables the assembly of large, complex metagenomes.  

\section{Results}

\subsection{Connectivity analysis of metagenome datasets}

\subsubsection{Presence of a single, highly-connected lump in all datasets}
We selected datasets from three diverse, medium to high complexity metagenomes from the human gut \cite{Qin:2010p189}, cow rumen \cite{Hess:2011p686}, and agricultural soil (unpublished) . We also included a simulated, error-free metagenome of a high complexity, high coverage (\textasciitilde{}10x) microbial community \cite{Pignatelli:2011p742} (Table 1). To study the effects of increased sequencing, we included two additional subsets of the agricultural soil metagenome containing 50 million and 100 million reads each.   The coverage of each metagenome was estimated by aligning sequencing reads to corresponding assembled contigs (see Methods).   The human gut had the highest coverage, estimated at 32.4\%.  The coverages of the cow rumen, small soil, medium soil, and large soil metagenomes were significantly less, 3.5\%, 1.4\%, 4.7\%, and 5.60\%, respectively (Table 1).  

The connectivity of reads within the assembly graph of each dataset was evaluated within a de Bruijn graph representation (see Methods).  As an initial step, reads contributing to disconnected portions of the assembly graph were separated.  For each metagenome, regardless of origin, we identified a dominant, highly-connected set of sequencing reads which we referred to as the "lump" (Figure 1, Table 1).  In the simulated dataset, this lump consisted of 5\% of the reads.  In the metagenomes, the size of the lump ranged from 7\% (in the smallest soil metagenome) to 67\% (in the human gut metagenome) of the total reads.  For the three soil datasets of increasing size, the size of the lump was disproportionately larger than increases in sequencing.  As the number of reads increased by 2-fold and 5-fold, the size of the lump increased by 5-fold and 14-fold, respectively.  

@AC - I chose to just leave this out but FYI...  In a smaller subset of the human gut metagenome (50 million reads), the relative size of the lump was similar and comprised over 75\% of the sequencing reads (data not shown). 

\begin{table}
\center{\includegraphics[width=5in]{./figures/table1.pdf}}
\caption{The connectivity of sequencing reads from medium to high complexity metagenomes from the soil, rumen, and human-gut were analyzed.  Read coverage was estimated by aligning sequencing reads to Velvet-assembled contigs (K=33).  A dominant lump, or largest disconnected component of each metagenome assembly graph, was identified in each metagenome. }
\end{table}

\begin{figure}
\center{\includegraphics[width=5in]{./figures/lump_sizes.pdf}}
\caption{The presence of a dominant lump was identified for all metagenomes studied, regardless of origin of the metagenome.}
\end{figure}

@Note that people always ask if these are highly repetitive - make some comments some where
@Note that filtering out highly abundant k-mers effects - CTB, need to analyze...need to ask question about this

\subsubsection{Characterizing assembly graphs in dominant, disconnected lumps}

We next assessed the degree of connectivity of sequences within theidentified lump of each metagenome.  The local graph density of reads within the de Bruijn assembly graph of each metagenome's lump was calculated (Table 2).  The local graph density is defined as the number of k-mers, or sequences of length k, found within a distance of N divided by N.  For a mixture of 112 complete bacterial genomes, fewer than 2\% of the nodes within the assembly graph had an average graph density greater than 20.   The simulated short reads generated from these genomes resulted in a lump made up of 4.7\% of total reads.  Within this lump, 17\% of the nodes had an average graph density greater than 20.  Similarly, the average graph density in lumps from the environmental metagenome datasets ranged from 21 to 50\%.  

\begin{table}
\center{\includegraphics[width=5in]{./figures/density_table.pdf}}
\caption{The number of highly dense k-mer nodes with local assembly graph densities greater than 20 (N=100) for various metagenomes and a mixture of 112 reference genomes \cite{Pignatelli:2011p742}. For the human gut metagenome, partial traversal of the graph (82\%) was used to calculate the number of highly dense nodes due to computational limitations.  The local graph density (N=10) for all nodes in each metagenome was calculated for each k-mer position within reads (see Figure X), and the average density for all k-mer positions is shown.}  
\end{table}

We next determined the extent to which graph density varied by position along sequenced reads by measuring the average local graph density within ten steps of every k-mer by position in a read (Figure 2).  For the simulated dataset, the average local density was stable for all positions along reads.  In contrast, metagenome datasets had increased average local graph densities towards the 3'-end read position.  The reads in the lumps in the soil metagenomes had the most variability of local graph densities by position, with standard deviations ranging from 25\% - 35\% of the average density.

@is this a real number that people usually report, % of standard deviations, there's probably a better way to say this

\begin{figure}
\center{\includegraphics[width=5in]{./figures/density_pos.pdf}}
\caption{The extent to which average local graph density varies by read position is shown for the lump of various datasets.  Unlike the graph density of the simulated metagenome lump, average graph density of sequences in metagenomic lumps varied significantly by position along the read.}
\end{figure}

\subsection{Identification of highly-connecting k-mers and their effects on assembly}

\subsubsection{Characteristics of highly-connecting sequences in simulated and metagenomic lump reads}

We next identified sequences within each lump which were probable causes of high local graph density using a systematic assembly graph traveral algorithm (see Methods).  For environmental metagenomes, we identified 6 to 8\% of unique k-mers within each lump were sources of high connectivity, with the exception of the smallest soil metagenome lump which contained less than 1\%  (Table 3).  In contrast, the lump of the simulated metagenome contained fewer of these highly connective k-mers, approximately 3\% of total unique k-mers.  

Having identified these highly-connective sequences, we next assessed the extent to which these k-mers were found at specific positions along a sequencing read (Figure 3).  In the simulated metagenome lump, the highly-connecting k-mers did not exhibit any bias with regard to position within the read.  In other words, these k-mers had equal probability of being located at any position along the read.  In the case of metagenome lumps, highly connective k-mers were more prevalent at position-specific locations along the read.   For the three soil metagenomes, the fraction of total k-mers which were identified as highly-connective increased at the 3'-end of the read.  In the same read region in the human-gut and rumen metagenome lumps, the fraction of these k-mers decreased relative to other regions of the reads.    

\begin{figure}
\center{\includegraphics[width=5in]{./figures/position_read.pdf}}
\caption{The extent to which highly-connecting k-mers are present at specific positions within sequencing reads for various metagenomes.}
\end{figure}

\subsubsection{Characteristics of highly-connecting sequences in simulated and metagenomic lump assemblies}

We were interested in the incorporation of the highly-connective k-mers into the final assembly of the sequences within the lumps.  For all lumps, we found that the there was an enriched presence of these k-mers within reads compared to the final assembly (Table 3).  In the simulated lump, there was an 6-fold enrichment of these sequences in the reads compared to the assembled contigs.  Likewise, in metagenomic lumps, there was a 3 to 6 time enrichment of highly-connective k-mers in reads over the assemblies.  Examining the position of these highly-connecting k-mers within assembled contigs, we found that these sequences were being disproportionately placed on the ends of contigs in every metagenome assembly studied, including the simulated metagenome (Figure 4) .  


\begin{table}
\center{\includegraphics[width=5in]{./figures/stoptag_table.pdf}}
\caption{Highly-connecting k-mers were more highly enriched in sequencing reads compared to assembled contigs.  All metagenomes (except for the soil 3 and human gut) were assembled with Velvet (as described in Methods) with K=25, 27, 29, 31, 33.  Enrichment ratios for soil 3 and human gut metagenomes calcuated from assemblies at only at K=33 due to computational limitations.}
\end{table}

\begin{figure}
\center{\includegraphics[width=5in]{./figures/pos-spec-bias-contigs.pdf}}
\caption{When incorporated into an assembly, highly-connecting sequences (k-mers) were disproportionately present at the ends of contigs.}
\end{figure}

\subsection{Effects of removal of highly-connecting k-mers on assembly}

\subsubsection{Effects of removing highly-connecting k-mers on the simulated lump assembly}
To study the effects of the highly-connecting k-mers on assembly, we performed assemblies of the simulated metagenome lump with and without removal of the identified highly-connective k-mers from reads.  As a result of removing these sequences, several assembly statistics improved for the simulated data.  Whereas the unfiltered assembly contained 1,844 contigs (maximum size = 3,787 bp) with 1,365,436 bp, the filtered assembly contained 2,301 contigs (maximum size = 5,826 bp) with 1,733,645 bp (Table 4).  Overall, we found the unfiltered and filtered assemblies to be quite different.  Based on comparing constituent k-mer composition, we estimated that the assemblies are approximately 69\% similar.  Furthermore, of the 1,844 contigs assembled from the unfiltered assembly, we estimate that 375 (20\%) are unique.  In the filtered assembly, we estimate that 1,136 of the 2,301 contigs (49\%) are unique. 
	
We next identified protein coding regions from the simulated assemblies by predicting open reading frames (ORFs) from assembled contigs with lengths greater than 500 bp.  The filtered assembly contained more ORFs compared to the unfiltered assembly, 3,049 filtered ORFs and 2,401 unfiltered ORFs (Table 4).  To evaluate the accuracy of each assembly, we compared the predicted ORFs to the proteins within the original 112 genomes which were used to generate the simulated metagenome.  In the unfiltered assembly, 2,395 (99.8\%) ORFs matched the reference genomes while in the filtered lump assembly, more assembled ORFs, 3,037 (99.6\%), matched the reference genomes.  Overall, the unfiltered and filtered assemblies shared shared a total of 2,018 ORFs which varied in lengths between unfiltered and filtered assemblies.  Comparing the alignment length of predicted ORFs to reference proteins, we found that the filtered assembly had slightly longer ORF alignment lengths compared to the unfiltered assembly (460 ORFs), and few ORFs had longer alignments in the unfiltered assembly (402 ORFs).  The large majority of shared ORFs had no difference in alignment lengths (1,156 ORFs) (Figure 5).  

@@are the unique orfs shorter? - small signal, not sure if good enough if strong enough to include here, decided to leave this out
@ Note that I left out the edge effects of ORFs...I thought it was just too funky and vague.  I think the annotations against the references make for the stronger argument.  We can put it back in though if you like it.

\begin{table}
\center{\includegraphics[width=5in]{./figures/assemblies.pdf}}
\caption{Comparison of unfiltered and filtered (removal of highly connecting k-mers) assemblies of various metagenome lumps.}
\end{table}

\begin{figure}
\center{\includegraphics[width=5in]{./figures/orf_lengths.pdf}}
\caption{Alignment lengths of ORFs shared between unfiltered and filtered assemblies (removal of highly connective k-mers) of the simulated metagenome are compared.  Difference in alignment length of predicted ORFs to contigs is shown (filtered assembly - unfiltered assembly alignment length).}
\end{figure}

\subsubsection{Effects of removing highly-connecting k-mers on metagenome lumps}
Comparing the unfiltered and filtered assemblies of the rumen, human gut, and soil metagenomes, we observed that removing highly-connecting k-mers consistently resulted in increased maximum contig lengths (ranging from 48\% to 157\%).   The effects of filtering highly-connecting k-mers on the total length of assembly and number of predicted ORFs varied (Table 4).   For the smallest soil, human gut, and rumen metagenomes, filtered assemblies had a longer total length.  For the soil metagenomes containing 100 million and 500 million reads, the total length of assembly decreased 6 and 17\%, respectively.  The total number of predicted ORFs decreased for filtered assemblies (ranging from 1 to 22\% fewer ORFs), with the exception of the smallest soil metagenome which had an increase in number of predicted ORFs.  Comparing the constituent k-mers between unfiltered and filtered assemblies, we observed that the assemblies of the environmental metagenomes, like the simulated assemblies, were also quite different.  Similarity, based on constituent k-mer comparisons, ranged from 58\% to 81\% for unfiltered and filtered assemblies.   A large proportion of contigs were also unique to either the unfiltered or filtered assembly.

We compared the assemblies of the ruman metagenomic lump to 15 draft genomes predicted from the published assembly of the full rumen metagenome \cite{Hess:2011p686}.  We compared the number of predicted ORFs from unfiltered and filtered assemblies which matched these genomes.  For the unfiltered assembly, 4,607/36,857 (12\%) predicted ORFs had a best match to the draft genomes while in the filtered assembly, more matches 4,618/36,210 (13\%) ORFs matched the genomes.  A total of 117 ORFs were identified in both assemblies.  Among these ORFs, 51 ORFs when aligned to draft genomes were longer in the filtered assembly, 50 ORFs were longer in the unfiltered assembly, and 16 ORFs had identical alignment lengths to draft genomes.  

\section{Discussion}

\subsection{Characteristics of the lump did not appear to be biological in origin.}

For each metagenome, the assembly graph was separated into into millions of disconnected assembly subgraphs representing sequences originating from different genomes.  The largest of these subgraphs, called the lump, contained a disproportionate number of reads relative to other subgraphs.  We initially considered that this lump consisted mainly of connecting sequences which were conserved across multiple genomes (i.e. 16S rRNA, ITS regions).  However, efforts to remove conserved genes from datasets did not significantly break apart the lump (data not shown).  Furthermore, the large size of the lump within metagenomic data compared to that of a simulated metagenomes (i.e., 75\% of reads in the human gut metagenome vs. 5\% in simulated) suggested that some connectivity within this lump was not biological (Table 1, Figure 1).  Comparing soil metagenomes of increasingly larger sizes, we observed that with increasing sequencing, there was a supra-linear increase in the size of the lump and thus increase in graph connectivity. These results combined suggested the presence of spurrious connectivity within metagenomic lumps, and we proceeded to further analyze the connectivity these sequences.

\subsection{Position-specific biases indicate that sequencing artifacts are present in the lump.}
We first assessed the degree of connectivity within the lump by measuring the local graph density of metagenomic lumps.  We compared local graph densities within metagenomic lumps to those of a mixture of whole genomes and the lump of simulated reads.  For a mixture of genomes, the local graph density was measured to be low with less than 2\% of nodes having a graph density greater than 20.  For the lump of connected reads of a simulated Illumina sequencing dataset of these genomes, the graph density was larger with 17\% of nodes having a graph density greater than 20.   For the metagenomic lumps, the degree of connectivity was consistently larger than the simulated dataset lump, with an average of 35\% of metagenomic nodes having a graph density of greater than 20.   In addition to this observed increased connectivity, we also identified position-specific biases of local graph densities within a read for all metagenomes studied (Figure 2).  As this bias should not be present in randomly sampled shotgun sequences and is not observed in the simulated dataset, spurrious connectivity clearly exists within the metagenomic sequences within the lump.  

To further explore the sources of this spurrious connectivity, we identified the highly-connecting sequences within each lump which were likely causes of the lump itself.  Similarly to the observed position-specific bias of local graph density along the read, the presence of these highly-connecting sequences also had a biased presence in locations within sequencing reads (Figure 3).  This bias supports the presence of non biological sequences within the metagenomic lumps.

\subsection{Comparing position-specific trends in various metagenomes indicates preferential attachment.}
Similar position-specific trends were observed for both the local graph density and the fraction of highly-connecting sequences within the metagenomic lumps of all samples studied (Figures 2 and 3).  The 3'-end biases of the two largest soil metagenomes were the most pronounced.  For these soil metagenome lumps, the size of the lumps and associated position-specific trends increased at greater rates than the amount of sequencing, further supporting the presence of sequencing artifacts.  We suspect that this connectivity is the result of an effect referred to as "preferential attachment" \cite{Barabasi:1999p1083}. In this case, highly connecting "X" sequences in a lump recruit a number of connecting "Y" reads into the lump.  As more sequences are added, these "Y" reads, which do not necessarily have to be highly-connective, recruit more "Z" reads into the lump resulting in increasingly larger lump size.  In the case of the studied metagenomes, 6-8\% of unique k-mers were identified to be highly-connnective.  In our example, these would be the "X" sequences, and the number of "Y" and "Z" sequences which are connected to these sequences would be affected by the coverage of the metagenome.  For soil metagenomes, where sequencing coverage is low and diversity is high (5.6\% coverage for the largest soil metagenome), increased sequencing would cause preferential attachment of "X", "Y", and "Z" reads resulting in increasingly larger lump sizes.  For metagenomes with less complexity, like that of the human gut (32.4\% coverage), the number of the "Y" and "Z" reads which are lump-associated but not highly-connective would increase at a greater rate than than "X" reads.   This would effectively introduce a greater proportion of sequences in the lump which would not be identified as highly-connective and result in an overall decrease in the total fraction of these sequences (as seen in the rumen and human gut metagenomes in Figure 3).  

\subsection{Removing highly-connecting sequences improves assembly overall.}
Although some of the highly-connecting sequences in metagenomic lumps are sequencing artifacts (given their position-specific bias), it is apparent that not all of these sequences are artifacts as they are also present in the error-free simulated dataset (3\% of the unique k-mers identified as highly-connective).  However, regardless of the origin of these sequences, we were interested in their incorporation into resulting assemblies.  All sequences within the simulated and metagenomic lumps were separately assembled.  We observed that the highly-connective sequences which we previously identified were under-represented in the final assembly compared to their presence in original sequencing reads (Table 3).  Moreover, when these sequences were incorporated into assembly, they tended to begin or end contigs (Figure 4).  These results suggest that, overall, the assemblers are challenged by characteristics of these sequences regardless of their origin and that the removal of these sequences should have little effect on subsequent assembly efforts. 

We evaluated the effects of removing these sequences with the simulated dataset.  This dataset is ideal for benchmarking because it contains no sequencing errors or biases and can be validated by its original reference genomes.  We compared the de novo assembly of the simulated dataset before and after filtering out highly-connected sequences to evaluate the biological effects of these sequences on assembly.  Overall, we found that the unfiltered and filtered assemblies were quite different based on constituent k-mer composition (31\% different) and the total number of unique ORFs (34\% ORFs unique to only the filtered assembly).  In general, the removal of the highly-connective sequences resulted in an overall improvement of the assembly.  The filtered assembly contained 29\% more contigs and 51\% longer assembly length (for contigs greater than 500 bp) than the unfiltered assembly (Table 4).  The filtered assembly also resulted in 27\% more predicted ORFs which had matches to reference genomes.  Additionally, when comparing similar ORFs between assemblies, the filtered assembly contained a greater number of equal length or longer ORFs than the unfiltered assembly (Figure 5).  In combination, these results indicate that removing highly-connecting sequences, regardless of their origin, improves the overall assembly.  

Comparing soil, human gut, and rumen metagenomes, we also observed large differences in unfiltered and filtered assemblies.  The removal of highly-connecting sequences resulted in gaining new contigs and losing unfiltered contigs (Table 4).  These metagenomic assemblies are more challenging to evaluate as the original source genomes are unknown.  Based on a small number of draft reference genomes which were previously validated in cite Hess, we evaluated the rumen unfiltered and filtered assemblies.  In comparing the predicted ORFs of the rumen assemblies, we found that the filtered assembly predicted slightly more ORFs compared to the unfiltered assembly.  Additionally, the rumen filtered assembly contained more ORFs which had improved alignments to the reference genomes.  
	
\section{Conclusion}

As datasets from NGS technologies continue to increase in size, our ability to analyze this sequencing data must reevaluated.  Here, we demonstrate that efforts to resolve components of the complex metagenome assembly graphs are bottlenecked by the presence of highly-connective sequences that are have both biological and artificial origins.  We show that in an error-free simulated dataset, the removal of these sequences (though biological in origin) improves the overall assembly.  We thus propose the identification and subsequent removal of these sequences from metagenomes.  This approach results in not only discarding sequences which we demonstrated to be artifactual but also allow one to break apart the dominant, highly-connected subgraph which contains these sequences.  As a consequence of being able to resolve this component of the assembly graph, de novo assembly can effiently be performed on the separate, smaller sub-graphs.  Overall, our efforts provide a better understanding of the connectivity of metagenomes and gives us the ability to accurately scale de novo assembly which is critical to for successful metagenomic analysis.  



\section{Methods}

\subsection{Metagenomic datasets}
All datasets, with the exception of the agricultural soil metagenome, were from previously published datasets. Rumen-associated sequences (Illumina) were randomly selected from the rumen metagenome available at ftp://ftp.jgi-psf.org/pub/rnd2/Cow\_Rumen \cite{Hess:2011p686}. Human-gut associated sequences (Illumina) of samples MH0001 through MH0010 were obtained from ftp://public.genomics.org.cn/BGI/gutmeta/Raw\_Reads \cite{Qin:2010p189}. The agricultural soil metagenome was from an Iowa corn soil metagenome and is currently unpublished. All reads used in this study were quality-trimmed for Illumina's read segment quality control indicator, where a quality score of 2 indicates that all subsequent regions of the sequence should not be used. After quality-trimming, only reads with lengths greater than 30 bp were retained. All quality trimmed reads used in this study are available at X.  The number of reads after quality-trimming is shown in Table 1 for each metagenome. The simulated high complexity, high coverage dataset was previously published (Pignatelli, 2011).  

Coverage of each metagenome was estimated by aligning trimmed sequencing reads to assembled contigs with lengths greater than 500 bp.  For coverage estimates, the assembly of each metagenome was performed using Velvet (v1.1.05) with the following parameters:  K=33, exp cov=auto, cov cutoff=0, no scaffolding.  For datasets larger than 50 million reads, metagenomes were partitioned with the methods described in (cite PNAS paper) prior to assembly to overcome computational memory limitations for assembly.  Reads were aligned to assembled contigs with Bowtie (v0.12.7), allowing for a maximum of two mismatches.  

\subsection{Lightweight, compressible de Bruijn graph representation}
We used a lightweight probabilistic de Bruijn graph representation to explore k-mer connectivity of the assembly graph (cite PNAS paper). The de Bruijn graph stores k-mer nodes in Bloom filters and keeps edges between nodes implicitly, i.e. if two k-mer nodes exist with a k-1 overlap, then there is an edge between them. Bloom filters are a probabilistic set storage data structure with false positives but no false negatives, thus the size of the bloom filters were selected to be appropriate for each dataset and the memory available.
	
For analyzing the graph connectivity of the studied datasets, we used 4 x 48e9 bit bloom filters for soil, rumen, and human gut datasets, and 4 x 1e9 bit bloom filters for the simulated dataset. As metagenomic sequencing contains a mixture of multiple organisms, we could exploit the biological structure of the sequencing by partitioning the assembly graph into disconnected subgraphs that represent the original DNA sequence components. The set of the largest number of reads which were connected in the assembly graph is referre to above as a single, highly-connected lump. 

\subsection{Local graph density and identifying highly-connected k-mers}
We implemented a systematic traversal algorithm to identify highly connected components of the assembly graph.  Waypoints were labeled to cover the graph such that they
are a minimum distance of L apart. Originating from a waypoint, all k-mers were systematically and exhaustively traversed within a region that is the distance N.   The local graph density was calculated as the number of X k-mers reachable within a distance N divided by the distance N.  Thus, the local graph density of a linear sequence would be 2, and additional branches or repeats would increase graph density.   To evaluate the extent of connectivity within each metagenomic lump, we calculated the number of nodes (K=32) with a local graph density greater than 20 when N=100 and determined these nodes to be "highly dense" (Table 2).  We also evaluated the degree to which graph density varied by position along a read.  For each k-mer in a read, the local graph density within N=10 nodes was calculated, and the total average local graph density by k-mer position for all reads was subsequently calculated (Figure 2).
@brain fart - fix highly dense - there's better phrases I'm sure

\subsection{Identifying properties of highly-connecting k-mers}
The enrichment of knot-causing k-mers in unfiltered reads was estimated by identifying the fraction of unique k-mers in unfiltered sequencing reads and in their corresponding assembled contigs. The ratio of these k-mer fractions (unfiltered reads/assembled contigs) was used to estimate the enrichment of knot-causing k-mers in the reads. 
	
\subsection{De Novo Metagenomic Assembly}
De novo metagenomic assembly of reads within each metagenomic lump was completed with Velvet (v1.1.02) with the following parameters: velveth -short -shortPaired (if applicable to the dataset) and velvetg -exp\_cov auto -cov\_cutoff 0 -scaffolding no \cite{Zerbino:2008p665}.  For soil 1, soil 2, rumen, and simulated metagenomes, assemblies were performed at K=25, 27, 29, 31, and 33 and merged with Minimus (Amos v3.1.0, \cite{Sommer:2007p1253}).  For soil 3 and human gut metagenomes, assemblies were performed at only K=33 due to the size of the datasets and memory limitations.  

To study the effects of removing highly connecting sequences on assembly, the identified highly connective k-mers were filtered from reads by truncating the reads at the location of these sequences and the remaining reads assembled as described above.  To compare unfiltered and filtered assemblies, we compared the total number of contigs, total assembly length, maximum contig size, and total number of predicted ORFs.  ORFs were predicted using Fraggenescan(v1.1.15) with the following parameters: -complete=0 -training=454\_10 \cite{Rho:2010p397}.  The k-mers which constituted the unfiltered and filtered assemblies were compared.  To calculate the number of shared unique k-mers between assembly A and assembly B, constituent k-mers of contigs from assembly A were loaded into bloom filters. Subsequently, the constituent k-mers from the assembly B were queried against the assembly A k-mers. The number of shared unique k-mers is dependent on which assembly was initially loaded into the bloom filters. Thus, each comparison was completed twice, once with the unfiltered assembly and once with the filtered assembly initially loaded into the bloom filter. Assembly similarity was determined by the lowest fraction of shared unique k-mers between these two comparisons (Figure X). 

@Need to add in vector assembly comparisons - not sure how this is done well enough to write about it, vector blah blah

@AC...Assemblies were also performed with ABYSS (v1.2.0, cite) with the following parameters: ABYSS -k 33  - do we want to anything with this?

The location of highly-connecting k-mers within assembled unfiltered contigs was examined by dividing each contig into 100 equally-sized regions.  The fraction of highly-connecting k-mers within each region was calculated for each metagenome and is shown in Figure 4.

To evaluate assemblies of the simulated and rumen datasets, ORFs and contigs were aligned to the original 112 genomes used to generate the simulated metagenome or 15 draft rumen genomes  \cite{Hess:2011p686} using BLAST (v2.2.25).   Mismatches between assembled sequences and reference genomes were identified to evaluate the accuracy of assembly.  To determine the effects of removing highly-connecting sequences, the alignment lengths of contigs from unfiltered and knot-filtered assemblies which shared identical top blast alignments were compared.    


\bibliography{artifacts-paper-bib}
\bibliographystyle{plain}


\end{document}